File: domain/src/main/java/top/contins/synapse/domain/entity/ScheduleType.java
package top.contins.synapse.domain.entity;

/**
 * 日程类型枚举
 */
public enum ScheduleType {
    MEETING("会议"),
    PERSONAL("个人"),
    WORK("工作"),
    STUDY("学习"),
    ENTERTAINMENT("娱乐");

    private final String displayName;

    ScheduleType(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
}

File: domain/src/main/kotlin/top/contins/synapse/domain/model/schedule/ScheduleType.kt
package top.contins.synapse.domain.model.schedule

enum class ScheduleType(val displayName: String, val defaultColor: Long) {
    MEETING("会议", 0xFF2196F3),      // 蓝色
    PERSONAL("个人", 0xFF4CAF50),     // 绿色
    WORK("工作", 0xFFFF9800),         // 橙色
    STUDY("学习", 0xFF9C27B0),        // 紫色
    ENTERTAINMENT("娱乐", 0xFFE91E63), // 粉色
    EVENT("事项", 0xFF607D8B)         // 蓝灰色
}
